M2/Macaulay2/packages/Msolve.m2

```diff
@@ -0,0 +1,527 @@
+msolvePresent := run ("type msolve >/dev/null 2>&1") === 0
+-- msolveVersion := if msolvePresent then ... -- no way to get version info!!!
+msolveVersionNeeded := "0.6.4"
+msolvePresentAndModern := msolvePresent -- and match("^[0-9.]+$",msolveVersion) and msolveVersion >= msolveVersionNeeded
+
+newPackage(
+	"Msolve",
+	Version => "1.24.05", 
+    	Date => "July 2024",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}, {Name => "Mike Stillman", 
+		  Email => "mike@math.cornell.edu", 
+		  HomePage => "https://math.cornell.edu/michael-e-stillman"},{Name => "Anton Leykin", 
+		  Email => "leykin@math.gatech.edu", 
+		  HomePage => "https://antonleykin.math.gatech.edu/"}},
+    	Headline => "An interface to the msolve package (https://msolve.lip6.fr/) which computes Groebner Basis and does real root isolation",
+    	AuxiliaryFiles => true,
+	DebuggingMode => false,       
+	OptionalComponentsPresent => msolvePresent
+	);
+msolveEXEwithOpts:=(opt)->"msolve -v "|opt#"level of verbosity"|" -t "|toString(opt#"number of threads");
+export{
+    "msolveGB",
+    "msolveSaturate",
+    "msolveEliminate",
+    "msolveRUR",
+    "msolveLeadMonomials",
+    "msolveRealSolutions"
+    }
+importFrom_Core { "raw", "rawMatrixReadMsolveFile" }
+
+use'readMsolveOutputFile := true;
+readMsolveOutputFile = method()
+readMsolveOutputFile(Ring,String) := Matrix => (R,mOut) -> if use'readMsolveOutputFile 
+    then map(R, rawMatrixReadMsolveFile(raw R, mOut)) else (
+	msolveOut := get mOut;
+    	moutStrL:=separate(",",first separate("[]]",last separate("[[]",msolveOut)));
+    	--the line below should be replaced by a call to the C-function to parse the string
+    	M2Out:=for s in moutStrL list value(s);
+    	matrix {M2Out};
+    	)
+
+inputOkay=method(TypicalValue=>Boolean);
+inputOkay(Ideal):=I->(
+    R:=ring I;
+    if not instance(R,PolynomialRing) then (print "input must be in a polynomial ring over a field";return false;);
+    kk:=coefficientRing(R);
+    if instance(kk,InexactFieldFamily) then(
+        print "input must be over the rationals or a (prime) finite field of chacteristic between 2^16 and 2^32";
+	return false;
+	);
+    if char(kk)>0 then (
+	if (char(kk)<2^16) or (char(kk)>2^32) then(
+	    print "input must be over the rationals or a (prime) finite field of chacteristic between 2^16 and 2^32";
+	    return false;
+	    );
+	);
+    return true;
+    );
+msolveGB=method(TypicalValue=>Matrix, Options=>{"level of verbosity"=>0, "number of threads"=>maxAllowableThreads});
+msolveGB(Ideal):=opt->I->(
+    if not inputOkay(I) then return 0;
```

Comment:
This should through an error. Then `try msolveGB I` can be added as a very succinct hook for `gb`.

Reply:
This is more difficult probably. Other interface packages simply throw an error when the program terminates with a nonzero return value (i.e. in this case there's no need to even parse the output).

Reply:
Yes, actually thinking about it I can just check the size of the output string. Since the msolve output file will be empty if and only if it crashes in the run. I can add this (today or maybe tomorrow).